views/data.ejs

```diff
@@ -153,6 +168,59 @@
                 Plotly.newPlot('distribution-plot', plot);
             });
 
+            $.ajax({url: "/api/report/module"})
+            .always(function (modules, textStatus) {
+                if (textStatus != "success") {
+                    bootbox.alert({ message: "Error fetching site data! - " + data.statusText, backdrop: true });
+                    return;
+                }
+
+                var data = [{
+                    values: modules.map(function(module) { return sum(Object.values(module.versions)); }),  // number of implementations of each module
+                    labels: modules.map(function(module) { return module.name }),   // module names
+                    type: 'pie'
+                }];
+
+                Plotly.newPlot('modules-piechart', data);
```

Comment:
While I admire your dedication to Plotly, I'm not sure a pie chart sliced into over a hundred 3% or smaller slices is useful. Instead, it might be more helpful to show the most popular modules in a list of decreasing popularity (name and percent usage). For example, list all modules used by two or more sites in decreasing order of percent usage. Make sure percent usage is calculated using number of sites with data as denominator (not total number of sites) -- i.e., if 4 sites out of 100 are running the atlas module and have sent data payloads and 2 of those use the foo module, then report foo usage as 50%, not 2%. I'd suggest using a div that shows the top ten modules and uses scroll for vertical overflow so someone could scroll within that div to see beyond the 10th most popular module.

Reply:
Updated the PR :) Here's your [portal](https://atlas-stg.openmrs.org/data) to the page.

Reply:
Nicely done! I'm surprised to see the EMR API Module at 84.7457627118644%; I really thought it would have been 84.7457627118645%. 😜 

Some additional suggestions now that I can see it on staging:
* Omit Atlas Module. It's kinda cheating, since it will – by definition – always be at 100%.
* Round percentages to integers and eliminate the space between number & percent sign (e.g., "EMR API Module - 85%").
* Make sure the modules are sorted by usage descending. They are roughly – but not exactly – sorted at the moment (e.g., "Form Entry App Module - 83%" is above "Reporting - 97%").
* Add a subtitle like "(data from n sites)"
* Use the term "Sites" instead of "Markers" ... you and I (and Google Maps API) knows them as markers, but the OpenMRS Community sees them as sites using OpenMRS
* Given that the Sites by Country map and Popular Modules are sexier and more interesting, let's not put them add the bottom. Instead, move the map up to the top alongside the number of sites, drop the "Sites by Country" title (it's implied), try to make the map a little bigger if you can, and move the legend closer or alongside the image. Then make the "Popular Modules" next and keep it the same width (half-sized) as the bar graphs. So, it comes out something like this:

![image](https://user-images.githubusercontent.com/860834/63469141-7bc0f780-c437-11e9-905f-807bbbddbde4.png)

Reply:
Made the changes :)

> Make sure the modules are sorted by usage descending. They are roughly – but not exactly – sorted at the moment (e.g., "Form Entry App Module - 83%" is above "Reporting - 97%").

Except this one. The list appears sorted for me.

Reply:
Looking much better!

It's strange. The sorting issue appears to be specific to Chrome. The modules are sorted fine in Safari, but they're still out of order in Chrome. I'm using Chrome Version 76.0.3809.100 (Official Build) (64-bit) on macOS and this is what I see:
![image](https://user-images.githubusercontent.com/860834/63612335-04f93b00-c5ac-11e9-853c-03a2289e3a61.png)

Also, could you add some css to center the site stats? I'm guessing something like this might work:
```css
site-stats: {
  position: relative;
}
site-stats > div {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: transformX(-50%) transformY(-50%);
}
```

Reply:
@bmamlin Woah, I'm to be having the same issue on Chrome. I'll have a have look.

Thanks for the CSS, I did try `vertical-align: center;` previously but gave up after it didn't work. :p

Reply:
@bmamlin The list appears sorted on Chrome too, now. Had to change the '>' to a '-' in the sort function.